# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/udev/scsi_dev_discs.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./extras/scsi-devfs.sh.orig	2006-01-04 17:02:54.370422160 +0100
+++ ./extras/scsi-devfs.sh	2006-01-04 17:04:39.350462768 +0100
@@ -83,6 +83,16 @@
 l_phy="bus/pci/$l_pci/scsi/$l_com"
 s_phy="$1/pci/$pci_addr/$s_com"
 
-echo $l_phy $s_phy $l_log $s_log
+for x in /dev/discs/disc* ; do
+       if readlink $x/disc | grep -q "$l_pci" ; then
+               x=`echo $x | cut -f3 -dc` # gives the number in disc0
+               break
+       fi
+       unset x
+done
 
+if [ -z "${x}" ] ; then
+	x="`ls /dev/discs/ | grep -c .`"
+fi
 
+echo $l_log $s_log discs/disc${x}/${lpart}
